providers/os/resources/npm/package_lock.go

```diff
@@ -10,22 +10,64 @@ import (
 	"go.mondoo.com/cnquery/v10/providers-sdk/v1/upstream/mvd"
 )
 
-type PackageJsonLockEntry struct {
-	Version string `json:"version"`
-	Dev     bool   `json:"dev"`
+// packageLock is the struct to represent the package.lock file
+// see https://docs.npmjs.com/cli/v10/configuring-npm/package-lock-json
+type packageLock struct {
+	Name            string `json:"name"`
+	Version         string `json:"version"`
+	LockfileVersion int    `json:"lockfileVersion"`
+	Requires        bool   `json:"requires"`
+	// Packages maps package locations to an object containing the information about that package,
+	// root project is typically listed with a key of ""
+	Packages map[string]packageLockPackage `json:"packages"`
+	// Dependencies contains legacy data for supporting versions of npm that use lockfileVersion: 1 or lower.
+	// We can ignore that for lockfileVersion: 2+
+	Dependencies map[string]packageLockDependency `jsonn:"dependencies"`
 }
 
-// PackageJsonLock is the struct to represent the package.lock file
-type PackageJsonLock struct {
-	Name         string                          `json:"name"`
-	Version      string                          `json:"version"`
-	Dependencies map[string]PackageJsonLockEntry `jsonn:"dependencies"`
+type packageLockDependency struct {
+	Version   string `json:"version"`
+	Resolved  string `json:"resolved"`
+	Integrity string `json:"integrity"`
+	Dev       bool   `json:"dev"`
 }
 
+type packageLockPackage struct {
+	Name      string             `json:"name"`
+	Version   string             `json:"version"`
+	Resolved  string             `json:"resolved"`
+	Integrity string             `json:"integrity"`
+	License   packageLockLicense `json:"license"`
+}
+
+type packageLockLicense []string
+
+// UnmarshalJSON is a custom unmarshaler for the packageLockLicense type. It allows to handle the license field
+// which could be either a string or an array.
+func (l *packageLockLicense) UnmarshalJSON(data []byte) (err error) {
```

Comment:
I might have missed it but I don't see a test file where licenses is an array of string, can we add such a test case?
I only see `"license": "Artistic-2.0"`

Reply:
I added a new test to cover the multiple licenses case